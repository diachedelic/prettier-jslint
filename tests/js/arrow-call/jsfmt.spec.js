const errors = {
  espree: ["class-property.js"],
};

run_spec(__dirname, ["babel", "flow", "typescript"], {
  tabWidth: 4,
  trailingComma: "none",
  bracketSpacing: false,
  arrowParens: "always",
  errors,
});
