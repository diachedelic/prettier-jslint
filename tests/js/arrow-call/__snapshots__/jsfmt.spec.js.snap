// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow_call.js - {"tabWidth":4,"trailingComma":"none","bracketSpacing":false,"arrowParens":"always"} format 1`] = `
====================================options=====================================
arrowParens: "always"
bracketSpacing: false
parsers: ["babel", "flow", "typescript"]
printWidth: 80
tabWidth: 4
trailingComma: "none"
                                                                                | printWidth
=====================================input======================================
const testResults = results.testResults.map(testResult =>
  formatResult(testResult, formatter, reporter)
);

it('mocks regexp instances', () => {
  expect(
    () => moduleMocker.generateFromMetadata(moduleMocker.getMetadata(/a/)),
  ).not.toThrow();
});

expect(() => asyncRequest({ url: "/test-endpoint" }))
  .toThrowError(/Required parameter/);

expect(() => asyncRequest({ url: "/test-endpoint-but-with-a-long-url" }))
  .toThrowError(/Required parameter/);

expect(() => asyncRequest({ url: "/test-endpoint-but-with-a-suuuuuuuuper-long-url" }))
  .toThrowError(/Required parameter/);

expect(() => asyncRequest({ type: "foo", url: "/test-endpoint" }))
  .not.toThrowError();

expect(() => asyncRequest({ type: "foo", url: "/test-endpoint-but-with-a-long-url" }))
  .not.toThrowError();

const a = Observable
  .fromPromise(axiosInstance.post('/carts/mine'))
  .map((response) => response.data)

const b = Observable.fromPromise(axiosInstance.get(url))
  .map((response) => response.data)

// func(
//   veryLoooooooooooooooooooooooongName,
//   veryLooooooooooooooooooooooooongName =>
//     veryLoooooooooooooooongName.something()
// );

promise.then(result => result.veryLongVariable.veryLongPropertyName > someOtherVariable ? "ok" : "fail");

=====================================output=====================================
const testResults = results.testResults.map((testResult) => formatResult(
    testResult,
    formatter,
    reporter
));

it("mocks regexp instances", () => {
    expect(() => moduleMocker.generateFromMetadata(
        moduleMocker.getMetadata(/a/)
    )).not.toThrow();
});

expect(() => asyncRequest({url: "/test-endpoint"})).toThrowError(
    /Required parameter/
);

expect(() => asyncRequest({
    url: "/test-endpoint-but-with-a-long-url"
})).toThrowError(/Required parameter/);

expect(() => asyncRequest({
    url: "/test-endpoint-but-with-a-suuuuuuuuper-long-url"
})).toThrowError(/Required parameter/);

expect(() => asyncRequest({
    type: "foo",
    url: "/test-endpoint"
})).not.toThrowError();

expect(() => asyncRequest({
    type: "foo",
    url: "/test-endpoint-but-with-a-long-url"
})).not.toThrowError();

const a = Observable.fromPromise(axiosInstance.post("/carts/mine")).map(
    (response) => response.data
);

const b = Observable.fromPromise(axiosInstance.get(url)).map(
    (response) => response.data
);

// func(
//   veryLoooooooooooooooooooooooongName,
//   veryLooooooooooooooooooooooooongName =>
//     veryLoooooooooooooooongName.something()
// );

promise.then((result) => (
    result.veryLongVariable.veryLongPropertyName > someOtherVariable
    ? "ok"
    : "fail"
));

================================================================================
`;
